sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/convert/MappingCosmosConverter.java

```diff
@@ -142,12 +165,83 @@ public JsonNode writeJsonNode(Object sourceEntity) {
             final String id = value == null ? null : value.toString();
             cosmosObjectNode.put("id", id);
         }
+        if (transientFields != null && !transientFields.isEmpty()) {
+            // Strip fields with @Transient annotation from the JsonNode so they are not persisted.
+            cosmosObjectNode = stripTransientFields(cosmosObjectNode, transientFields);
+        }
 
         mapVersionFieldToEtag(sourceEntity, cosmosObjectNode);
 
         return cosmosObjectNode;
     }
 
+    /**
+     * To repopulate any transient fields that were stripped from the original object
+     * @param <T> type of source entity
+     * @param transientValues transient fields and their values
+     * @param responseItem must not be {@literal null}
+     * @return CosmosItemProperties
+     * @throws MappingException no mapping metadata for entity type
+     * @throws CosmosAccessException fail to map document value
+     */
+
+    public <T> JsonNode repopulateAnyTransientFieldsFromMap(JsonNode responseItem, HashMap<Field, Object> transientValues) {
+        if (transientValues.isEmpty()) {
+            return responseItem;
+        }
+        ObjectNode updatedItem = (ObjectNode) responseItem;
+        for (Field field : transientValues.keySet()) {
+            Object value = transientValues.get(field);
+            if (value != null) {
+                updatedItem.set(field.getName(), objectMapper.valueToTree(value));
+            }
+        }
+        return updatedItem;
+
+    }
+
+    /**
+     * To get transient fields and their values
+     * @param <T> type of source entity
+     * @param object must not be {@literal null}
+     * @param transientFields transient fields
+     * @return HashMap
+     */
+    public <T> HashMap<Field, Object> getTransientFieldsAndValuesMap(T object, List<String> transientFields) {
```

Comment:
Its better to be generic in API params and return types. 
For example, the way you have `List<String>` as input and not `ArrayList<String>`
Similarly, the API should take and return type `Map<Field, Object>` instead of type `HashMap<...>` so that if in future you need to change the implementation, it won't break the public API contract.

Reply:
Here and other places in the code.

Reply:
Ok sure, will change that to make consistent.